src/ui/wxWidgets/TreeCtrl.h

```diff
@@ -54,42 +54,33 @@ typedef std::map<pws_os::CUUID, wxTreeItemId, std::less<pws_os::CUUID> > UUIDTIM
 
 class TreeCtrl;
 
-class TreeScrollTimer: public wxTimer
-{
-public:
-    // start scrolling 1/4 second (if the mouse hasn't been clicked)
-  enum { DELAY = 250 };
-    
-  TreeScrollTimer();
-  
-  void setOwner(TreeCtrl *owner) { m_owner = owner; };
-  bool Start() { return wxTimer::Start( TreeScrollTimer::DELAY, true ); };
-
-  virtual void Notify();
+typedef void (TreeCtrl:: *TreeCtrlMemberFncPtr)(void);
 
-private:
-  TreeCtrl *m_owner;
-
-  wxDECLARE_NO_COPY_CLASS(TreeScrollTimer);
-};
-
-class TreeCollapseTimer: public wxTimer
+class TreeCtrlTimer: public wxTimer
 {
 public:
+  enum {
+    // start scrolling 1/4 second (if the mouse hasn't been clicked)
+    DELAY_SCROLLING = 250,
     // start Collapse or Expand after 1.5 second (if the mouse hasn't been clicked/moved)
-  enum { DELAY = 1500 };
-    
-  TreeCollapseTimer();
+    DELAY_COLLAPSE = 1500
+  };
+  
+  TreeCtrlTimer(TreeCtrl *aowner, TreeCtrlMemberFncPtr acallback, int atime, bool aonetime = true) : m_owner(aowner), m_callback(acallback), m_time(atime), m_one_time(aonetime) { }
+  TreeCtrlTimer(TreeCtrlMemberFncPtr acallback, int atime, bool aonetime = true) : m_owner(nullptr), m_callback(acallback), m_time(atime), m_one_time(aonetime) { }
```

Comment:
The compiler issues many warnings about the initialization order of the timer-related members.

Reply:
> * The compiler issues many warnings about the initialization order of the timer-related members.
Next check in solves the issues about -Wreorder
> * Do we need the second constructor and TreeCtrlTimer::setOwner ?
I do not need this, but left for later deriving a general timer class from this code part.